src/functions/definitions/base_function.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ManifestFunctionSchema,
+  ManifestFunctionType,
+} from "../../manifest/manifest_schema.ts";
+import {
+  ParameterSetDefinition,
+  PossibleParameterKeys,
+} from "../../parameters/types.ts";
+import { FunctionDefinitionArgs, ISlackFunctionDefinition } from "../types.ts";
+import { SlackManifest } from "../../manifest/mod.ts";
+
+export abstract class FunctionDefinition<
+  InputParameters extends ParameterSetDefinition,
+  OutputParameters extends ParameterSetDefinition,
+  RequiredInput extends PossibleParameterKeys<InputParameters>,
+  RequiredOutput extends PossibleParameterKeys<OutputParameters>,
+> implements
+  ISlackFunctionDefinition<
+    InputParameters,
+    OutputParameters,
+    RequiredInput,
+    RequiredOutput
+  > {
+  public id: string;
+  abstract type: ManifestFunctionType;
+
+  constructor(
+    public definition: FunctionDefinitionArgs<
+      InputParameters,
+      OutputParameters,
+      RequiredInput,
+      RequiredOutput
+    >,
+  ) {
+    this.id = definition.callback_id;
+    this.definition = definition;
+  }
+
+  registerParameterTypes(manifest: SlackManifest) {
+    const { input_parameters: inputParams, output_parameters: outputParams } =
+      this.definition;
+    manifest.registerTypes(inputParams?.properties ?? {});
+    manifest.registerTypes(outputParams?.properties ?? {});
+  }
```

Comment:
`registerParameterTypes` is intended to find any types used by a function and register them to the manifest if they're not already there. This seems like something we'd only need to do on `SlackFunction`

Reply:
Ahh I see this makes a lot of sense 💯  let me try out removing this from the base class

Reply:
I figured out why I implemented `registerParameterTypes` at the a abstract class level 💡the reason for this is that  `addStep` expects [ISlackFunctionDefinition](https://github.com/slackapi/deno-slack-sdk/blob/156def3d584b6b66a8e3935761b508188d8494fe/src/functions/types.ts#LL302C30-L302C30) as an [argument](https://github.com/slackapi/deno-slack-sdk/blob/156def3d584b6b66a8e3935761b508188d8494fe/src/workflows/mod.ts#L119)

We can potentially change this but it will be a bigger refactor then anticipated, LMK what you think

Reply:
The `deno-slack-runtime` filters out the functions of `type: API` from the manifest

Reply:
Edit this was in the [deno-slack-hooks build.ts](https://github.com/slackapi/deno-slack-hooks/blob/main/src/build.ts#L32)

But I've modified the SDK so `connectors` are not exported in the manifest